### PR TITLE
[PRO-162] "PopUp" Component

### PR DESCRIPTION
**Pull request by Timmehs**
Introduces the PopUp component which normalizes our modal styles. It's pretty flexible since it exposes the full API of the [react-bootstrap modal](https://react-bootstrap.netlify.app/docs/components/modal#modal), while setting some useful defaults.

I began this branch by organizing our bootstrap scss files a bit, and getting styles closer to our end-goal purely with variable overrides.

image

image

image

(This one passes a bodyClass="px-4" for more horizontal padding)
image

Because of the positioning of modals, I found it tricky to write storybook stories for this, but I think the usages so far provide good prior-art examples for how to use the PopUp for various purposes.

### CODE REVIEW

File: src/components/LoginModal/constants.ts [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+
 export const LOGIN_PAGES = {
   SIGN_IN: 0,
   SIGN_UP: 1,
   FORGOT_PASSWORD: 2,
 }
+
+// This should return localised string keys, e.g. `loginModal.signUp.title` once i18n is
+// implemented. For now we're hardcoding strings.
+export const PAGE_TITLES: IPopUpTitle[] = [
```
Comment (eddie-m-m) [inline src/components/LoginModal/constants.ts:11]: Are you going to be adding `../PopUp/PopUpTitle`? I don't see it in the files pushed up.

Comment (Timmehs) [reply src/components/LoginModal/constants.ts]: Good catch, removed.

**Review by eddie-m-m — COMMENTED** [review]
I think it looks and feels great! All functionality is intact.

The only thing I wonder about is the close button on the modals. It feels superfluous in many instances (e.g. there are only two options on the delete modal: cancel or delete. So why the extra close button which kind of looks out of place style-wise). Does it have to be kept on them?

**Comment by Timmehs** [comment]
> I think it looks and feels great! All functionality is intact.
> 
> The only thing I wonder about is the close button on the modals. It feels superfluous in many instances (e.g. there are only two options on the delete modal: cancel or delete. So why the extra close button which kind of looks out of place style-wise). Does it have to be kept on all of them?
@eddie-m-m Nope, the success modal shows an example of PopUp without a close button. It is optional and wont render unless `closeButton` is passed to PopUp

**Review by eddie-m-m — APPROVED** [review]
Nice slam dunk! Really streamlines modal usage well.

**Pull request merged**